lib/processors/nonAsciiEscaper.js

```diff
@@ -0,0 +1,99 @@
+ * @type {number}
+ */
+const CHAR_CODE_OF_LAST_ASCII_CHARACTER = 127;
+
+// use memoization for escapeUnicode function for performance
+const memoizeEscapeUnicodeMap = {};
+const memoizeEscapeUnicode = function(sChar) {
+	if (memoizeEscapeUnicodeMap[sChar]) {
+		return memoizeEscapeUnicodeMap[sChar];
+	}
+	memoizeEscapeUnicodeMap[sChar] = escapeUnicode(sChar);
+	return memoizeEscapeUnicodeMap[sChar];
+};
+
+/**
+ * Escapes non ASCII characters with unicode escape sequences.
+ *
+ * @see https://en.wikipedia.org/wiki/ASCII
+ * @see https://tools.ietf.org/html/rfc5137#section-6.1
+ *
+ *
+ * @param {string} string input string with non ascii characters, e.g. L♥VE
+ * @returns {{string: (string), modified: boolean}} output string with all non ascii
+ * characters being escaped by unicode sequence, e.g. L\u2665VE
+ */
+const escapeNonAscii = function(string) {
+	let result = "";
+	let modified = false;
+	for (let i = 0; i < string.length; i++) {
+		const char = string[i];
+		// check for non ascii characters (characters which have a char code
+		// greater than the ascii character code range)
+		if (string.charCodeAt(i) > CHAR_CODE_OF_LAST_ASCII_CHARACTER) {
+			result += memoizeEscapeUnicode(char);
+			modified = true;
+		} else {
+			result += char;
+		}
+	}
+	return {
+		modified,
+		string: result
+	};
+};
+
+/**
+ * Escapes non ASCII characters with unicode escape sequences.
+ *
+ * @example
+ * const encoding = nonAsciiEscaper.getEncodingFromNiceName("ISO-8859-1");
+ * nonAsciiEscaper({resources, options: {encoding}});
+ *
+ *
+ * @public
+ * @alias module:@ui5/builder.processors.nonAsciiEscaper
+ * @param {Object} parameters Parameters
+ * @param {module:@ui5/fs.Resource[]} parameters.resources List of resources to be processed
+ * @param {Object} [parameters.options] Options
+ * @param {string} [parameters.options.encoding="utf8"] resource file encoding (node.js based encodings). Use #getEncodingFromNiceName to get the encoding string
+ * {@link https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings Node.js character encodings};
+ * @returns {Promise<module:@ui5/fs.Resource[]>} Promise resolving with the processed resources
+ */
+module.exports = async function nonAsciiEscaper({resources, options={}}) {
+	const encoding = options.encoding || "utf8";
+
+	async function processResource(resource) {
+		const resourceString = (await resource.getBuffer()).toString(encoding);
+		const escaped = escapeNonAscii(resourceString);
+		// only modify the resource's string if it was changed
+		if (escaped.modified) {
+			resource.setString(escaped.string);
+		}
+		return resource;
+	}
+
+	return Promise.all(resources.map(processResource));
+};
+
+const encodingMap = {
+	"UTF-8": "utf8",
+	"ISO-8859-1": "latin1",
+};
+
+/**
+ * @param {string} encoding encoding labels: "UTF-8" and "ISO-8859-1"
+ * @returns {string} node.js character encoding string, e.g. utf8 and latin1
+ */
+module.exports.getEncodingFromNiceName = function(encoding) {
```

Comment:
Doesn't this function fit much better in the task?
See @RandomByte's earlier comment.

It's never called in the processor itself.
I think it is cleaner to keep the processor solely node-based, as it only understands the node char encodings anyway.

This fits much better into the task, which relies on external encodings like used by UI5.
It's also only called by the task so...

Also:
Maybe call it getEncodingNameFromAlias() ?

Reply:
@Thodd **the place**: if this method `getEncodingFromNiceName` is within the processor it can be easier consumed from the server and doesn't require a task.
@Thodd **the name**: getEncodingNameFromAlias is also good. I like it even better.

Reply:
@matz3 @tobiasso85 and myself talked about this yesterday. We came to the agreement that while it fits better in the task (because it bridges between the UI5- and node-native world), its something you need when calling the processor. So it's more convenient for the UI5 Server and the bundler tasks to have it available there.